src/core/ucc_progress_queue_mt.c

```diff
@@ -63,15 +66,28 @@ static void ucc_pq_mt_dequeue(ucc_progress_queue_t *pq,
 
 static int ucc_pq_mt_progress(ucc_progress_queue_t *pq)
 {
-    int              n_progressed = 0;
+    int              n_progressed =  0;
+    double           timestamp    = -1;
     ucc_coll_task_t *task;
     ucc_status_t     status;
+
     pq->dequeue(pq, &task);
     if (task) {
         if (task->progress) {
             task->progress(task);
         }
         if (UCC_INPROGRESS == task->super.status) {
+            if (UCC_COLL_TIMEOUT_REQUIRED(task->args)) {
```

Comment:
ucc_unlikely() ?

Reply:
i don't think this is right hint. E.g., for the application which actually uses TM (FB) it will always produce wrong hint.